test/web_integration.dart

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) 2015, Google Inc. Please see the AUTHORS file for details.
+// All rights reserved. Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+library dartpad.web_integration;
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:http_server/http_server.dart';
+import 'package:unittest/unittest.dart';
+import 'package:webdriver/webdriver.dart';
+import 'package:which/which.dart';
+
+/**
+ * - try and determine the best webdriver client to use based on the environment
+ *   and system
+ * - start a web server on `build/web/`
+ * - start a webdriver client (chromedriver, phantomjs, saucelabs)
+ * - run the integration tests, providing the tests each their own new webdriver
+ *   client and running on a fresh dartpad page
+ */
+main(List<String> args) async {
+  List<DriverFactory> factories = [
+    new SauceLabsDriverFactory(),
+    new ChromeDriverFactory(),
+    new PhantomJSDriverFactory(),
+  ];
+
+  DriverFactory factory;
+
+  for (DriverFactory f in factories) {
+    if (f.isAvailable) {
+      factory = f;
+      break;
+    }
+  }
+
+  if (factory == null) {
+    print('''
+No webdriver candidates found. Either set up the env. variables for using
+saucelabs, or install chromedriver or phantomjs.
+
+See https://github.com/dart-lang/dart-pad/blob/master/doc/webdriver.md
+for more information.
+''');
+    exit(1);
+  }
+
+  _setupTests(factory);
+}
+
+/* WebDriver tests */
+
+// Test that the title is correct - that it starts with "DartPad".
+testCheckTitle(WebDriver driver) async {
+  String title = await driver.title;
+  expect(title, startsWith('DartPad'));
+}
+
+testCheckTitleExact(WebDriver driver) async {
+  String title = await driver.title;
+  expect(title, 'DartPad (β)');
+}
+
+/* WebDriver plumbing */
+
+_setupTests(DriverFactory factory) async {
+  print('using ${factory} webdriver');
+
+  MicroServer server = await MicroServer.start(path: 'build/web', port: 8888);
+  print('serving ${server.path} from ${server.urlBase}');
+
+  await factory.startFactory();
+
+  var cleanup = () async {
+    print('closing server');
+    await server.destroy();
+    return factory.stopFactory();
+  };
+
+  group('integration', () {
+    int count = 0;
+
+    WebDriver driver;
+
+    setUp(() async {
```

Comment:
I think we setup and teardown the driver for each test.We have this hack in so that after the group finishes, we can teardown the driver factory and the server.

I'd definitely be interested in a less hacky way to listen for group completion, and for overall test run completion.

Glad to hear this got you up and running! I'm currently trying to get webdriver working with saucelabs and sauce_connect, and running into a bunch of issues :(